src/Language/PureScript/Interactive/IO.hs

```diff
@@ -21,9 +32,42 @@ onFirstFileMatching f pathVariants = runMaybeT . msum $ map (MaybeT . f) pathVar
 -- Locates the node executable.
 -- Checks for either @nodejs@ or @node@.
 --
-findNodeProcess :: IO (Maybe String)
-findNodeProcess = onFirstFileMatching findExecutable names
-  where names = ["nodejs", "node"]
+findNodeProcess :: IO (Either String String)
+findNodeProcess = onFirstFileMatching findExecutable ["nodejs", "node"] <&>
+  maybe (throwError "Could not find node.js. Do you have node.js installed and available in your PATH?") pure
+
+findNodeVersion :: String -> IO (Maybe String)
+findNodeVersion node = do
+  result <- readProcessWithExitCode node ["--version"] ""
+  return $ case result of
+    (ExitSuccess, version, _) -> Just version
+    (ExitFailure _, _, _) -> Nothing
+
+readNodeProcessWithExitCode :: Maybe FilePath -> [String] -> String -> IO (Either String (ExitCode, String, String))
+readNodeProcessWithExitCode nodePath nodeArgs stdin = runExceptT $ do
+  process <- maybe (ExceptT findNodeProcess) pure nodePath
+  (_, minor, _) <- lift (findNodeVersion process) >>= \case
+    Nothing -> throwError "Could not find node.js version."
+    Just version -> do
+      let semver = do
+            void $ char 'v'
+            major : minor : patch : _ <- fmap (read @Int) (many1 digit) `sepBy` void (char '.')
+            pure (major, minor, patch)
+      case parse (semver <?> "Could not parse node.js version.") "" version of
+        Left err -> throwError $ show err
+        Right (major, minor, patch)
+          | major < 12 -> throwError "Unsupported node.js version."
+          | otherwise -> pure (major, minor, patch)
+  let nodeArgs' = if minor < 7 then "--experimental-modules" : nodeArgs else nodeArgs
```

Comment:
Is this logic correct? The current logic reads that Node 12.5, 13.5, 14.2, 17.5, etc. would all fail this check.

Reply:
Resolved by https://github.com/working-group-purescript-es/purescript/pull/11